src/backend/distributed/commands/table.c

```diff
@@ -346,8 +345,15 @@ PostprocessAlterTableSchemaStmt(Node *node, const char *queryString)
  * function returns NIL.
  */
 List *
-PreprocessAlterTableStmt(Node *node, const char *alterTableCommand)
+PreprocessAlterTableStmt(Node *node, const char *alterTableCommand,
+						 ProcessUtilityContext processUtilityContext)
 {
+	if (processUtilityContext == PROCESS_UTILITY_SUBCOMMAND)
```

Comment:
can we move this check into / close to `AlterTableDefinesFKeyBetweenPostgresAndNonDistTable `?